src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -681,6 +682,18 @@ private static IPackageFile FindFileInPackage(string filePath, IEnumerable<IPack
             return null;
         }
 
+        private void ValidateFilesUnique(IEnumerable<IPackageFile> files)
+        {
+            var set = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (string file in files.Where(t => t.Path != null).Select(t => PathUtility.GetPathWithDirectorySeparator(t.Path)))
+            {
+                if (!set.Add(file))
+                {
+                    throw new PackagingException(NuGetLogCode.NU5133, string.Format(CultureInfo.CurrentCulture, NuGetResources.FoundDuplicateFile, file));
```

Comment:
We throw an exception when the first duplicate file has been found. Can we collect all the duplicates and throw the exception so that customer can fix all the issues at once before attempting 'pack' again?

Reply:
@kartheekp-ms How do you propose I actually fix this error?

Reply:
@jzabroski- The comment has already been addressed. Here is the link to the code. 

https://github.com/NuGet/NuGet.Client/blob/5353034fed272cb81fbe60326f334b99871d8b74/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L693-L703

Reply:
I figured it out - the issue can occur if a custom nuspec file has ambiguous targets.

Reply:
@kartheekp-ms I guess my observation is that you're only letting people know the target duplicates, not which source set the duplicates came from.  In my case, the unexpected source of duplicates was that a custom nuspec file had the duplicates.  I thought the actual source was from PackageFiles.

To be honest, I am still a bit confused by this unification of custom nuspecfile and PackageFiles.